### PR TITLE
#7 - add synchronization and importers

### LINKED ISSUE #18: [Providers] Add Bolt provider
The aim of this task is to import Bolt provider to the application.
Link to provider's site: https://bolt.eu

### LINKED ISSUE #19: [Providers] Add Lime provider
The aim of this task is to import Lime provider to the application.
Link to provider's site: https://www.li.me/locations

### LINKED ISSUE #22: [Providers] Add Spin provider
The aim of this task is to import Spin provider to the application.
Link to provider's site: https://www.spin.app

### LINKED ISSUE #23: [Providers] Add Helbiz provider
The aim of this task is to import Helbiz provider to the application.
Link to provider's site: https://helbiz.com/cities

### LINKED ISSUE #25: [Providers] Add Dott provider
The aim of this task is to import Dott provider to the application.
Link to provider's website: https://ridedott.com

### LINKED ISSUE #27: [Providers] Add Quick provider
The aim of this task is to import Quick provider to the application.
Link to provider's site: https://quick-app.eu/lokalizacje

### LINKED ISSUE #45: Add CitySeeder
The aim for this issue is to add real cities inside database that will further help importers in work.

### PR DESCRIPTION
**Pull request by vojcc**
It's connected with #7.
This should: close #18, close #19, close #22, close #23, close #25, close #27 and close #45.

* added few importers which follow the same scheme of scrapping and saving data,
* added jobs which are in queue so no 504 error will occur,
* added tables to inform about import status,
* running DataImporterService in schedule() in console Kernel.

I'm thinking of creating seeder not only for countries but for cities also. 
It would help Importers in work because some provider sites don't use country names (e.g. Lime) and as a result some cities are inserted to "city_without_couuntry_assigned" table and wait for administrator to assign them "manually".

I found API which has that data - https://countriesnow.space/ and I think it would be nice to get cities at least from "top" countries like France, Germany, USA etc.

Work is in progress because it seems that "iso2" is not always unified...

356171394_818768283155356_2358782449443495803_n
356068668_705363208266849_852542737984384551_n

To run importers:
```
1. Click "Run importers" button in navigation
2. php artisan queue:work --queue="importers"
```

### CODE REVIEW

**Comment by vojcc** [comment]
Fetchnig data from json file improves seeder execution speed. 

Added country capitals help Importers with scraping data:
* without capitals: 445 rows,
* with capitals: 455 rows.

image

File: app/Console/Kernel.php [hunk 1]
```diff
@@ -4,21 +4,20 @@
...
 {
-    /**
-     * Define the application's command schedule.
-     */
     protected function schedule(Schedule $schedule): void
     {
+        $schedule->call(function (): void {
+            $service = new DataImporterService();
+            $service->run("server");
+        })->everyMinute();
```
Comment (kamilpiech97) [inline app/Console/Kernel.php:18]: Import every minute is necessary?

**Comment by kamilpiech97** [comment]
conflicts

File: app/Console/Kernel.php (hunk 1 above)
Comment (vojcc) [reply app/Console/Kernel.php:18]: It's for testing to see if it works at all.

Comment (mtracz) [reply app/Console/Kernel.php:18]: You can use command to test scheduler.
`php artisan schedule:test`

More commands:
`php artisan schedule --help`

**Review by kamilpiech97 — APPROVED** [review]

**Review by EwelinaLasowy — COMMENTED** [review]
When I run command `php artisan migrate:fresh --seed` I see that no countries are seeded

image

(It will be cool, when it will be here some counter how many countries are seeded fine)

Because of that I can't check synchronization :(

File: database/seeders/DatabaseSeeder.php [hunk 2]
```diff
@@ -15,15 +13,12 @@ public function run(): void
...
+            CitiesAndCountriesSeeder::class,
         ]);
 
-        City::factory()->count(20)->create();
-        CityAlternativeName::factory()->count(10)->create();
-
         $this->call([
             ProviderListSeeder::class,
-            ProviderSeeder::class,
+            CodeSeeder::class,
```
Comment (EwelinaLasowy) [inline database/seeders/DatabaseSeeder.php:21]: I think you can use only one `$this->call...`

**Review by EwelinaLasowy — APPROVED** [review]

**Review by PiotrFedak — APPROVED** [review]

**Pull request merged**